### PR TITLE
Add support for HIP_PLATFORM=clang

### PR DESCRIPTION
**Pull request by amdkila**
- I haven't tested this yet...please don't merge

### CODE REVIEW

**Comment by saadrahim** [comment]
Did you see pr116

**Comment by saadrahim** [comment]
Did you see pr116

**Comment by amdkila** [comment]
I just saw it now. I suspect the user made the one change that was necessary to allow rocRAND to build. However, tests and packaging are a different story

**Comment by mihirparadkar** [comment]
@amdkila that is correct. The changes in #116 are more of a call-for-help since I have minimal experience with cmake (those changes require manual edits to Makefiles after the cmake command to build/run the benchmark and CuPy random array creation). These changes (plus adding gfx1010/1011/1012 to AMDGPU_TARGETS) seem like the correct way to fix my problem.

I'm happy to test this change for Linux Mint 19.3 and RX 5700XT + RX 5500XT.

**Comment by amdkila** [comment]
This ended up passing CI and fixes the clang issues. It should be safe to merge.

File: library/src/hiprand/hiprand_clang.cpp [hunk 1]
```diff
@@ -0,0 +1,382 @@
+// Copyright (c) 2017 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (saadrahim) [inline library/src/hiprand/hiprand_clang.cpp:1]: What is this file for?
I always bug people about copyright dates.

Comment (amdkila) [reply library/src/hiprand/hiprand_clang.cpp:1]: Good question. So, there's a part of the code where a file with hiprand_${HIP_PLATFORM} is used to read a bunch of hiprand functions that differ from hip and nvcc. In the library/src/hiprand directory, there are currently 2 files- hiprand_hcc.cpp and hiprand_nvcc.cpp. 

I just copied and pasted the hcc file and renamed it to _clang.cpp, so we can have both covered. I know, not an elegant solution, but it's the only one I figured out. I will fix the copyright dates.

Comment (mihirparadkar) [reply library/src/hiprand/hiprand_clang.cpp:1]: Just curious, is there anything HCC-specific in hiprand_hcc.cpp/hiprand_hcc.hpp ? In #116 I just made cmake point to hiprand_hcc.* when `${HIP_PLATFORM}` is `clang`

Comment (saadrahim) [reply library/src/hiprand/hiprand_clang.cpp:1]: No there is isn't, so your comment is noted that we shouldn't have two files.

Comment (saadrahim) [reply library/src/hiprand/hiprand_clang.cpp:1]: @amdkila don't bother with copyright dates, we should be able to have one file.

**Comment by saadrahim** [comment]
@amdkila Can we close this?

**Pull request closed without merging**